span-stacktrace/src/main/java/io/opentelemetry/contrib/stacktrace/StackTraceAutoConfig.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.stacktrace;
+
+import com.google.auto.service.AutoService;
+import io.opentelemetry.sdk.autoconfigure.spi.AutoConfigurationCustomizer;
+import io.opentelemetry.sdk.autoconfigure.spi.AutoConfigurationCustomizerProvider;
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigProperties;
+import io.opentelemetry.sdk.trace.ReadableSpan;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.time.Duration;
+import java.util.function.Predicate;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.annotation.Nullable;
+
+@AutoService(AutoConfigurationCustomizerProvider.class)
+public class StackTraceAutoConfig implements AutoConfigurationCustomizerProvider {
+
+  private static final Logger log = Logger.getLogger(StackTraceAutoConfig.class.getName());
+
+  private static final String CONFIG_MIN_DURATION =
+      "otel.java.experimental.span-stacktrace.min.duration";
+  private static final Duration CONFIG_MIN_DURATION_DEFAULT = Duration.ofMillis(5);
+
+  private static final String CONFIG_FILTER = "otel.java.experimental.span-stacktrace.filter";
+
+  @Override
+  public void customize(AutoConfigurationCustomizer config) {
+    config.addTracerProviderCustomizer(
+        (providerBuilder, properties) -> {
+          long minDuration = getMinDuration(properties);
+          Predicate<ReadableSpan> filter = getFilterPredicate(properties);
+          providerBuilder.addSpanProcessor(new StackTraceSpanProcessor(minDuration, filter));
```

Comment:
I don't think that allowing the construction of the `StackTraceSpanProcessor` with a negative duration makes sense, because it is always No-Op.

I would suggest to change `StackTraceSpanProcessor` to only allow min-durations greater or equal zero.
Then, here in the auto configuration only construct and add the processor if the duration is non-negative.

Reply:
good point, I've fixed that in d1f984a